at_client/src/main/java/org/atsign/client/util/KeysUtil.java

```diff
@@ -18,68 +18,104 @@
+        } else {
+            System.out.println("Saving keys to " + file.getAbsolutePath());
+            // noinspection ResultOfMethodCallIgnored
             file.delete();
         }
 
         String selfEncryptionKey = keys.get(selfEncryptionKeyName);
 
         Map<String, String> encryptedKeys = new TreeMap<>();
 
-        // We encrypt all the keys with the AES self encryption key (which is left unencrypted)
+        // We encrypt all the keys with the AES self encryption key (which is left
+        // unencrypted)
         encryptedKeys.put(selfEncryptionKeyName, selfEncryptionKey);
-        encryptedKeys.put(pkamPublicKeyName, EncryptionUtil.aesEncryptToBase64(keys.get(pkamPublicKeyName), selfEncryptionKey));
-        encryptedKeys.put(pkamPrivateKeyName, EncryptionUtil.aesEncryptToBase64(keys.get(pkamPrivateKeyName), selfEncryptionKey));
-        encryptedKeys.put(encryptionPublicKeyName, EncryptionUtil.aesEncryptToBase64(keys.get(encryptionPublicKeyName), selfEncryptionKey));
-        encryptedKeys.put(encryptionPrivateKeyName, EncryptionUtil.aesEncryptToBase64(keys.get(encryptionPrivateKeyName), selfEncryptionKey));
+        encryptedKeys.put(pkamPublicKeyName,
+                EncryptionUtil.aesEncryptToBase64(keys.get(pkamPublicKeyName), selfEncryptionKey));
+        encryptedKeys.put(pkamPrivateKeyName,
+                EncryptionUtil.aesEncryptToBase64(keys.get(pkamPrivateKeyName), selfEncryptionKey));
+        encryptedKeys.put(encryptionPublicKeyName,
+                EncryptionUtil.aesEncryptToBase64(keys.get(encryptionPublicKeyName), selfEncryptionKey));
+        encryptedKeys.put(encryptionPrivateKeyName,
+                EncryptionUtil.aesEncryptToBase64(keys.get(encryptionPrivateKeyName), selfEncryptionKey));
 
         String json = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(encryptedKeys);
         Files.write(file.toPath(), json.getBytes(StandardCharsets.UTF_8));
     }
 
-    private static File getKeysFile(AtSign atSign) {
-        return new File(System.getProperty("user.dir") + "/keys/" + atSign + "_key.atKeys");
-    }
-
     public static Map<String, String> loadKeys(AtSign atSign) throws Exception {
-        _makeRootFolder();
-        File file = getKeysFile(atSign);
-        if (! file.exists()) {
-            throw new AtException("loadKeys: No file at " + file.getAbsolutePath());
+        // check first if file exists at
+        // ~/.atsign/keys/$atSign_key.atKeys
+        File file = new File(rootFolder);
+        file = getKeysFile(atSign, rootFolder);
+
+        if (!file.exists()) {
+            // if keys do not exist in project dir, search in ~/
+            String keysFileName = System.getProperty("user.home") + "/.atsign/keys/";
+            file = getKeysFile(atSign, keysFileName);
+            // if file does not exist at
+            // ./keys/$atSign_key.atKeys
+            // Create dir to store key files
+            if (!file.exists()) {
+                _makeRootFolder();
```

Comment:
You could throw an exception here saying you couldn't find the file in either location. Also there's no need to make the root folder here.

Reply:
I had added _makeRootFolder() here as this is after having checked project dir, and the !file.exists() would be ~/. I was thinking that after searching both locations, that's when it would create the root folder